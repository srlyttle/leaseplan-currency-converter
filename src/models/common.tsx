export interface CurrencyResponse {
  rates: Object;
  base: string;
  date: string;
}

export interface CurrencyProps {
  // valueChanged: (event: React.ChangeEvent<HTMLInputElement>) => void;
  //removeCurrency: (event: React.MouseEvent<HTMLInputElement>) => void;
  currencyDetails: CurrencyDetails;
}
export interface CurrencyItemProps {
  removeCurrency: (event: React.MouseEvent<HTMLInputElement>) => void;
  currencyDetails: CurrencyDetails;
  currencyValue: string;
}

export interface CurrencyListProps {
  currencies: CurrencyResponse;
}
export interface CurrencyDetails {
  valueToBase: string;
  isBaseCurrency: boolean;
  currencyCode: string;
  currencyFlag: string;
  currencySymbol: string;
  currencyName: string;
  currencySummary: string;
  latestRate: number;
  lastDate?: string;
  baseCalcCurrency?: string;
  displayForUser: boolean;
}
