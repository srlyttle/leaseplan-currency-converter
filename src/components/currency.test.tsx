import React from 'react';
import { render } from '@testing-library/react';
import { screen, fireEvent } from '@testing-library/dom';
import { Currency } from './currency.component';
import { CurrencyDetails } from '../models/common';

describe('<Currency/>', () => {
  test('should render correct input for canadian dollar', () => {
    const currencyDetails: CurrencyDetails = {
      currencyCode: 'CAD',
      currencyFlag: 'http://www.geonames.org/flags/l/ca.gif',
      currencyName: 'Canadian Dollar',
      currencySummary: '1 CAD = 1.551 EUR',
      currencySymbol: '$',
      displayForUser: false,
      isBaseCurrency: false,
      latestRate: 1.551,
      valueToBase: '1.5510',
    };
    const { getByText, getByTestId, container } = render(
      <Currency currencyDetails={currencyDetails} />
    );
    //screen.debug();
    //screen.getByText("USD - US Dollar");
    expect(getByText('1 CAD = 1.551 EUR')).toBeInTheDocument();

    expect(getByTestId('testid-currency-summary')).toHaveTextContent(
      currencyDetails.currencySummary
    );
    expect(getByTestId('testid-currency-value-CAD')).toHaveTextContent(
      currencyDetails.valueToBase
    );

    expect(container).toMatchSnapshot();
  });
});
