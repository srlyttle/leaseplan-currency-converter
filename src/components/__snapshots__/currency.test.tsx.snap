// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Currency/> should render correct input for canadian dollar 1`] = `
<div>
  <div
    class="currency-card"
    id="CAD"
  >
    <div
      class="currency-value"
    >
      <div
        class="currency-symbol"
        data-testid="testid-currency-symbol"
      >
        $
      </div>
      <div
        class="currency-value-to-base"
        data-testid="testid-currency-value-CAD"
      >
        1.5510
      </div>
    </div>
    <div
      class="currency-summary"
      data-testid="testid-currency-summary"
    >
      1 CAD = 1.551 EUR
    </div>
  </div>
</div>
`;
